File: Programming Basics/04-For Loop - Exercise/cleverLily.js
function cleverLily(input) {

    let age = Number(input[0]);
    let washingMachinePrice =Number(input[1]);
    let priceForToy = Number(input[2]);

    let money = 0;
    let toys = 0;
    let multiplier = 1;

    for(let i = 1; i <= age; i++) {
        if(i % 2 === 0) {
            money += 10 * multiplier;
            multiplier++;
            money--;
        } else {
            toys++
        }
    }

    money += priceForToy * toys;

    let positivSum = Math.abs(money - washingMachinePrice);

    if(money >= washingMachinePrice) {
        console.log(`Yes! ${positivSum.toFixed(2)}`);
    } else {
        console.log(`No! ${positivSum.toFixed(2)}`);
    }
    
}   

cleverLily(["21",
"1570.98",
"3"])






//10
//20 = 30
//30 = 60
//40 = 100
//50 = 150

File: Programming Basics Java/src/ForLoopExercise/CleverLily.java
package ForLoopExercise;

import java.util.Scanner;

public class CleverLily {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int age = Integer.parseInt(scanner.nextLine());
        double priceForWashMachine = Double.parseDouble(scanner.nextLine());
        int priceForToy = Integer.parseInt(scanner.nextLine());

        int increment = 10;
        int savings = 0;

        for (int i = 1; i <= age; i++) {
            if (i % 2 == 0) {
                savings += increment;
                savings--;
                increment += 10;
            } else {
                savings += priceForToy;
            }
        }

        double diff = Math.abs(savings - priceForWashMachine);

        if (savings >= priceForWashMachine) {
            System.out.printf("Yes! %.2f", diff);
        } else {
            System.out.printf("No! %.2f", diff);
        }
    }
}
